docs/content/docs/data-structures/task-manager.md

```diff
@@ -1,5 +1,135 @@
 ---
 title: TaskManager
-status: unwritten
-draft: true
+menuPosition: 9
 ---
+
+## Introduction
+
+FluidFramework is designed to facilitate real-time collaboration in modern web applications by distributing data throughout its clients with the help of its many distributed data structures (DDSes). However, TaskManager uniquely distributes tasks rather than a dataset. Furthermore, TaskManager is designed to distribute tasks that should be exclusively executed by a single client to avoid errors and mitigate redundancy.
+
+{{% callout note "What exactly is a \"task\"?" %}}
+A task is simply code that should only be executed by **one** client at any given time. This could be as small as a single line of code, or an entire system . However, we reccomend large processes to frequently synchronize their progress in the case of an unexpected disconnection so another client can resume the process with minimal data loss.
+{{% /callout %}}
+
+### Task Queue
+TaskManager's main role is to maintain a queue of clients for each unique task. The client at the top of the queue is assigned the task, and is given permission to exclusively execute the task. All other clients will remain in queue until they leave, disconnect (unexpectedly), or the task is completed by the assigned client. It's important to note that TaskManager maintains the consensus state of the task queue. This means that locally submitted ops will not affect the queue until the op is accepted by all other clients. To learn more about conensus based data structures, click [here]({{< relref "./overview.md#consensus-data-structures" >}}).
+
+### Consensus Based DDS
+An important note about TaskManager is that it is a consensus based DDS. This essentially means that ops are not accepted until every client acknowledge and accepts the op. This differs from an "optimistic" DDS (i.e. [SharedMap]({{< relref "./map.md" >}})) which immediately accept ops and then relays them to other clients. For more information regarding different types of DDSes, click [here]({{< relref "./overview.md" >}}).
+
+## Usage
+
+### APIs
+
+The `TaskManager` object provides a number of methods to manage the execution of tasks. Please note: each API requires an input of `taskId` which is type `string`.
+
+
+- `volunteerForTask(taskId)` -- Adds the client to the task queue **once**. It returns a promise that resolves `true` if the client is assigned the task and `false` if the task was completed by another client. It will throw an error if the client disconnects while in queue.
+- `subscribeToTask(taskId)` -- Will continuously add the client to the task queue. Does not return a value, and will therefore require lisnening to [events](#events) to determine if the task is assigned, lost, or completed.
```

Comment:
We need an explanation of what kind of scenarios call for `volunteerForTask` and what kind call for `subscribeToTask`. And at least one concrete example of each. If not in this article, then somewhere else that is linked from here.